tool/src/main.rs

```diff
@@ -54,12 +57,39 @@ fn main() {
         tas = Tas::new().unwrap();
         println!("TAS created successfully.");
     }
-
-    let script_file = env::args().skip(1).next().unwrap_or("tas.lua".to_string());
+    let mut script_file = String::new();
+    if env::args().collect::<Vec<String>>().len() == 1 {
+        let cur_dir = std::env::current_dir().unwrap();
+        let cur_dir = cur_dir.to_str().unwrap();
+        let result = nfd::open_file_dialog(Some("lua"), Some(cur_dir)).unwrap_or_else(|e| { panic!(e) });
+        match result {
+            Response::Okay(file_path) => script_file = Path::new(&file_path).file_name().unwrap().to_str().unwrap().to_string(),
```

Comment:
Why do you change this to only use the file name? What if I select a file inside a folder somewhere completely different? Then you try to open a file with the same name but in the current directory. You should just return the full path here.